src/components/date-picker/date-picker.tsx

```diff
@@ -195,26 +149,61 @@ export class DatePicker {
                     helperText={this.helperText}
                     required={this.required}
                     value={this.formattedValue}
-                    onChange={this.handleChange}
+                    onChange={this.handleChangeFromInput}
+                    onFocus={this.showPortalElement}
+                    ref={(el) => (this.textField = el)}
                 />
+                <limel-portal
+                    containerId={this.portalId}
+                    visible={this.showPortal}
+                    containerStyle={portalContainerStyle}
+                >
+                    <limel-flatpickr-adapter
+                        type={this.type}
+                        value={this.value}
+                        onReady={this.hidePortalElement}
```

Comment:
It seems like the ready event isn't always emitted from all datepickers…

I'm wondering if there's some other way of getting the size of the portal set correctly… 🤔

Reply:
A potential but hacky fix to this came to my mind.

What if we trigger a "fake" ready event from a function wrapped in a `setTimeout` that always times out after X <time-units> unless an actual ready event is dispatched from flatpickr. (When/if the ready event is emitted from flatpickr we cancel the timeout). Doing that will ensure that we always set the portal to "not visible".

Reply:
The thought of that kind of makes my skin crawl… 😄

Sure we can try. But I'm worried we're building ourselves a real can of worms here. I think a fix like that is likely to introduce some very intermittent bugs.

But, flatpickr has an inline-mode, where the picker is just rendered inline, without any of flatpickr's own menu guff… maybe that could be helpful. It's also possible to not have it open when the user clicks the input, so we can instead catch that click, open the portal, and then open flatpickr.

I don't know how that might mess with the native-picker support on mobile though.